core/src/main/java/software/amazon/awssdk/core/auth/DefaultCredentialsProvider.java

```diff
@@ -63,9 +63,9 @@ private static AwsCredentialsProviderChain createChain(Builder builder) {
                 SystemPropertyCredentialsProvider.create(),
                 EnvironmentVariableCredentialsProvider.create(),
                 ProfileCredentialsProvider.create(),
-                ElasticContainerCredentialsProvider.builder()
-                                                   .asyncCredentialUpdateEnabled(builder.asyncCredentialUpdateEnabled)
-                                                   .build(),
+                ContainerCredentialsProvider.builder()
+                                            .asyncCredentialUpdateEnabled(builder.asyncCredentialUpdateEnabled)
+                                            .build(),
```

Comment:
Sorry that this is probably a dumb question. Why remove the `ElasticContainerCredentialsProvider` from the default chain in favor of the `ContainerCredentialsProvider`? It feels weird that we have a credentials provider in our default chain that customers can't directly use themselves.

Reply:
This was a rename - I had forgotten to remove `ElasticContainerCredentialsProvider` and the underlying `EC2CredentialsProvider` that it relied on - these are now both bundled into `ContainerCredentialsProvider` which handles getting credentials for any type of AWS container (EC2, ECS, Greengrass)

(I can make `ContainerCredentialsProvider` public again, it was you who'd suggested it be private... 😄 )

Reply:
I don't mind it being public. It wasn't marked either way, and the fact that the builder was package protected made me assume you meant for it to be internal.